pkg/controller/sparkapplication/controller.go

```diff
@@ -403,19 +391,25 @@ func (c *Controller) syncSparkApplication(key string) error {
 	appToUpdate := app.DeepCopy()
 	var updatedApp *v1alpha1.SparkApplication
 	switch app.Status.AppState.State {
-	case "", v1alpha1.NewState, v1alpha1.FailedState:
-		if shouldSubmit(app) {
-			glog.V(2).Infof("Creating Submission for SparkApp: %s", key)
-			attempt := app.Status.Attempts + 1
-			updatedApp = c.submitSparkApplication(appToUpdate, attempt)
+	case v1alpha1.NewState, v1alpha1.FailedState, v1alpha1.CompletedState:
```

Comment:
why are the states handled in 2 places? can we not put all the logic in once place?

Reply:
done